docarray/array/storage/memory/getsetdel.py

```diff
@@ -41,16 +40,16 @@ def _set_doc_by_id(self, _id: str, value: 'Document'):
         self._data[old_idx] = value
         self._id2offset[value.id] = old_idx
 
+    @needs_id2offset_rebuild
     def _set_docs_by_slice(self, _slice: slice, value: Sequence['Document']):
         self._data[_slice] = value
-        self._rebuild_id2offset()
 
+    @needs_id2offset_rebuild
     def _set_doc_value_pairs(
         self, docs: Iterable['Document'], values: Iterable['Document']
     ):
```

Comment:
why does this need rebuilding the offset ?

Reply:
no need, it is fixed